MP-APS/ResourceManager.cpp

```diff
@@ -72,62 +84,155 @@ unsigned int ResourceManager::LoadHDRI(const std::string_view path) const {
 }
 
 /***********************************************************************************/
-unsigned int ResourceManager::LoadTexture(const std::string_view path, const bool useMipMaps, const bool useUnalignedUnpack) {
+auto buildTextureCachePath(const std::filesystem::path& filenameNoExt) {
+	const std::filesystem::path filename{ filenameNoExt.string() + ".bin" };
+	return std::filesystem::path( COMPRESSED_TEX_DIR / filename);
+}
+
+/***********************************************************************************/
+void saveCompressedImageToDisk(const std::filesystem::path& target, const CompressedImageDesc& desc) {
+	if (!std::filesystem::exists(COMPRESSED_TEX_DIR)) {
+		if (!std::filesystem::create_directories(COMPRESSED_TEX_DIR)) {
+			std::cerr << "Failed to create texture cache directory: " << COMPRESSED_TEX_DIR << '\n';
+			return;
+		}
+	}
+
+	std::ofstream out(target, std::ios::binary);
+	if (out) {
+		out.write((char*)(&desc.size), sizeof(CompressedImageDesc::size));
+		out.write((char*)(&desc.width), sizeof(CompressedImageDesc::width));
+		out.write((char*)(&desc.height), sizeof(CompressedImageDesc::height));
+		out.write((char*)(&desc.format), sizeof(CompressedImageDesc::format));
+		out.write((char*)(desc.data.get()), desc.size);
```

Comment:
Use c-style cast since `reinterpret_case` cannot cast away `const` and I'd rather not nest a `const_cast` in here since it's very verbose